src/CheckboxSelector/widget/checkboxselector.js

```diff
@@ -1,459 +1,418 @@
+				this.boxWidthUnit = "";
+			} else if (this.boxWidthUnit == "u0025") {
+				this.boxWidthUnit = "%";
+			}
+		},
+
+		getAssocObjs: function (context) {
+			this.context = context;
+			var contextGUID = context.getGuid();
+
+			var nameSplit = this.nameAttr.split("/");
+			this.assoc = nameSplit[0];
+			this.referredEntity = nameSplit[1];
+
+			this.readOnlyBool = context.isReadonlyAttr(this.assoc);
+
+			if (this.readOnlyBool == true || this.readonly == true || this.isDisabled == true)
+				this.isInactive = true;
+			else
+				this.isInactive = false;
+
+			this.unsubscribe(this.changeHandler);
+			this.unsubscribe(this.changeHandler2);
+			this.unsubscribe(this.validationHandler);
+			this.changeHandler = this.subscribe({ guid: contextGUID, attr: this.assoc, callback: lang.hitch(this, this.changeReceived) });
+			this.changeHandler2 = this.subscribe({ guid: contextGUID, callback: lang.hitch(this, this.objChangeReceived) });
+			this.validationHandler = this.subscribe({ guid: contextGUID, val: true, callback: lang.hitch(this, this.validationUpdate) });
+
+			var xpath = "//" + this.referredEntity + this.constraint.replace(/\[\%CurrentObject\%\]/gi, contextGUID);
+
+			mx.data.get({
+				'xpath': xpath,
+				'filter': {
+					'limit': this.limit + "",
+					'sort': [[this.sortAttr, this.sortOrder]]
+				},
+				'error': function () {
+					logger.error("Checkbox Selector Widget: Retrieve objects failed.");
+				},
+				'callback': lang.hitch(this, this.renderList)
+			});
+
+		},
+
+
+		renderList: function (objs) {
+			domConstruct.empty(this.domNode);
+			var selectedObjs = this.context.getReferences(this.assoc);
+
+			var surrDiv = domConstruct.create('div', { 'class': 'mendixDataGrid_tablePane' });
+			var table = domConstruct.create('table', { 'class': 'mendixDataGrid_gridTable table-bordered', 'cellPadding': '0px', 'cellSpacing': '0px' }, surrDiv);
+
+			if (this.useHeaders) {
+				var thead = domConstruct.create('thead', { 'class': 'mendixDataGrid_gridHead' });
+				var headTr = domConstruct.create('tr', { 'class': 'mendixDataGrid_tableRow' }, thead);
+				var checkall = '';
+				if (this.hasSelectAll) {
+					checkall = domConstruct.create('input', { type: 'checkbox', 'class': 'checkbox_checkall' });
+					var checkallBool = (selectedObjs.length == objs.length);
+					checkall.checked = checkallBool;
+					if (this.isInactive) {
+						domAttr.set(checkall, "disabled", "disabled");
+					}
+				}
+
+				var checkboxHeadTh = domConstruct.create('th',
+					{ 'class': 'mendixDataGrid_tableHead mendixDataGrid_tableHeadFirst' });
+				var div = domConstruct.create('div', { 'class': 'mendixDataGrid_headContainer' }, checkboxHeadTh);
+				domConstruct.create('div', { 'class': 'mendixDataGrid_sortIcon' }, div);
+				var lastDiv = domConstruct.create('div', { 'class': 'mendixDataGrid_columnCaption' }, div);
+				lastDiv.append(checkall);
+
+				domStyle.set(checkboxHeadTh, 'width', this.boxWidth + this.boxWidthUnit);
```

Comment:
Do not remove any logic!!!
`this.boxWidth &&` missing.

Reply:
Reverted